test/subgraph.cpp

```diff
@@ -110,6 +131,56 @@ int test_main(int, char*[])
         }
     }
 
+    // This block generates a complete graph with 8 vertices,
+    // and puts the first and last four of the vertices into two children.
+    // Do these again to the children, so there are 4 grandchildren with 2 vertices for each.
+    // Use the copy constructor to generate a copy and compare with the original one.
```

Comment:
What about subgraph edges?  Those also need testing.

Reply:
I only checked the numbers. How do I compare two edge descriptors from two different graphs?

Reply:
You could load the structurally identical graphs with each node holding a unique ID in that graph identical to the equivalent edge in the other graph.

Reply:
Ok, I just added a comparison of two corresponding edge indices